### DISCUSSION TITLE
What Features Should I Build Next?

### DISCUSSION THREAD

**Discussion by summitt** [factionsecurity/faction#21] [Q&A]
Trying to prioritize my free time to support this project can be a challenge at times. I have a bunch of ideas that I think would be really cool to add but I can get in tunnel vision which creates a project over-tuned to my needs and not as useful to the community. 

So let me ask you all!... What features are must-haves for you that Faction does not currently include or maybe includes but poorly?

**Comment by jakuta-tech** [comment]
Hi Josh, thanks for this project.  I think it has a lot to offer over other pentest report builders.  One thing that I found was that the report generation probably wasn't as flexible as what I needed.  I was looking for something where I could add variables to the project and then have those populate into the report.   Those variables should be flexible enough so that I can create and name them for each project and store them in the database for later use on any other project.  Things like "contact_name1" "contact_name2" so that I could add those to the project and they could be populated into the report where the placeholders are.  I also wanted to create my own sections in the report (eg under the Detailed Summary / Risk Analysis text area) but wasn't able to add any extra sections other than the ones provided.  Also being able to add different templates for those sections depending on the client and job type.  Each of the vulnerabilities was also difficult to structure specific requirements around. I wanted to add these headings for each vuln but found it difficult to create them and then spent too much time updating the report to fit the requirements.

Description of the Vulnerability:
Vulnerability Advisory References:
Affected Locations: 
Impact: 
Proof of Concept:
Remediation: 
Additional Resources: 

Basically more flexibility for creating customised sections in the application, which can be added to the report if required would be great.

The other was a proper backup/export solution for client and template data.  It runs smooth on a Debian container in Proxmox and updates / bug fixes are great and really appreciated but each time I was rebuilding the app and starting again with a new container.

**Reply by summitt** [reply]
I definitely think there are many features that could be added to make reporting more flexible. I have plans as well to make configurable sections that could be added to the reporting and configurable vulnerability types (i.e. app sec vulns in one section, wireless issues in another section, social eng issues. .. etc). 

As for the backups you mentioned, you should be able to keep Mongo on a sep server and just deploy the Faction War file without needing to start all over again.  You could back Mongo and redeploy it and everything should work seamlessly.
